File: code/gms-backend/src/main/java/io/github/gms/secure/service/EventService.java
package io.github.gms.secure.service;

import io.github.gms.common.abstraction.GmsService;
import io.github.gms.secure.dto.EventListDto;
import io.github.gms.secure.dto.PagingDto;
import io.github.gms.secure.model.UserEvent;

/**
 * @author Peter Szrnka
 * @since 1.0
 */
public interface EventService extends GmsService {

	void saveUserEvent(UserEvent event);
	
	EventListDto list(PagingDto dto);
}

File: code/gms-frontend/src/app/components/event/service/event-service.ts
import { HttpClient } from "@angular/common/http";
import { Injectable } from "@angular/core";
import { Observable, map, tap } from "rxjs";
import { environment } from "../../../../environments/environment";
import { ServiceBase } from "../../../common/components/abstractions/service/service-base";
import { Paging } from "../../../common/model/paging.model";
import { getHeaders } from "../../../common/utils/header-utils";
import { EventList } from "../model/event-list.model";
import { Event } from "../model/event.model";

/**
 * @author Peter Szrnka
 */
@Injectable({ providedIn: 'root' })
export class EventService extends ServiceBase<Event, EventList> {

    constructor(http : HttpClient) {
        super(http, "event");
    }

    listByUserId(paging: Paging, userId?: number): Observable<Event[]> {
        return this.http.get<EventList>(environment.baseUrl + 'secure/event/list/' + userId + `?direction=${paging.direction}&property=${paging.property}&page=${paging.page}&size=${paging.size}`,
             { withCredentials: true, headers : getHeaders() })
            .pipe(tap(), map(value => value.resultList));
    }

    getUnprocessedEventsCount(): Observable<number> {
        return this.http.get<number>(`${environment.baseUrl}secure/event/unprocessed`, { withCredentials: true, headers : getHeaders() });
    }
}